opencti-platform/opencti-front/src/private/components/settings/Labels.jsx

```diff
@@ -1,256 +1,170 @@
-import withRouter from '../../../utils/compat-router/withRouter';
+import { usePaginationLocalStorage } from '../../../utils/hooks/useLocalStorage';
+import { useFormatter } from '../../../components/i18n';
+import DataTable from '../../../components/dataGrid/DataTable';
+import useQueryLoading from '../../../utils/hooks/useQueryLoading';
 
-const styles = () => ({
+const useStyles = makeStyles(() => ({
   container: {
     margin: 0,
-    padding: '0 200px 50px 0',
+    padding: '0 200px 0 0',
+    maxWidth: '100%',
   },
-});
+}));
 
 const LOCAL_STORAGE_KEY = 'Labels';
 
-class Labels extends Component {
-  constructor(props) {
-    super(props);
-    const params = buildViewParamsFromUrlAndStorage(
-      props.navigate,
-      props.location,
-      LOCAL_STORAGE_KEY,
-    );
-    this.state = {
-      sortBy: propOr('value', 'sortBy', params),
-      orderAsc: propOr(true, 'orderAsc', params),
-      searchTerm: propOr('', 'searchTerm', params),
-      view: propOr('lines', 'view', params),
-      numberOfElements: { number: 0, symbol: '' },
-      selectedElements: null,
-      deSelectedElements: null,
-      selectAll: false,
-    };
+const labelsLinesQuery = graphql`
+  query LabelsLinesPaginationQuery(
+    $search: String
+    $count: Int!
+    $cursor: ID
+    $orderBy: LabelsOrdering
+    $orderMode: OrderingMode
+    $filters: FilterGroup
+  ) {
+    ...LabelsLines_data
+    @arguments(
+      search: $search
+      count: $count
+      cursor: $cursor
+      orderBy: $orderBy
+      orderMode: $orderMode
+      filters: $filters
+    )
   }
+`;
 
-  saveView() {
-    saveViewParameters(
-      this.props.navigate,
-      this.props.location,
-      LOCAL_STORAGE_KEY,
-      this.state,
-    );
+const linesFragment = graphql`
+  fragment LabelsLines_data on Query
+  @argumentDefinitions(
+    search: { type: "String" }
+    count: { type: "Int", defaultValue: 25 }
+    cursor: { type: "ID" }
+    orderBy: { type: "LabelsOrdering", defaultValue: value }
+    orderMode: { type: "OrderingMode", defaultValue: asc }
+    filters: { type: "FilterGroup" }
+  )
+  @refetchable(queryName: "LabelsLinesRefetchPaginationQuery"){
+    labels(
+      search: $search
+      first: $count
+      after: $cursor
+      orderBy: $orderBy
+      orderMode: $orderMode
+      filters: $filters
+    ) @connection(key: "Pagination_labels")
+    {
+      edges {
+        node {
+          id
+          entity_type
+          ...LabelsLine_node
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        globalCount
+      }
+    }
   }
+`;
 
-  handleSearch(value) {
-    this.setState({ searchTerm: value }, () => this.saveView());
+const lineFragment = graphql`
+  fragment LabelsLine_node on Label {
+    id
+    entity_type
+    value
+    color
+    created_at
   }
+`;
 
-  handleSort(field, orderAsc) {
-    this.setState({ sortBy: field, orderAsc }, () => this.saveView());
-  }
+const Labels = () => {
```

Comment:
I can click on the label line and it redirects me to a an url with `null`
Do we have a prop to customize/disable this redirection?

Reply:
Simply forgot the actions 🥲
Fixed.